src/thing_list.c

```diff
@@ -304,8 +304,18 @@ long near_thing_pos_thing_filter_is_enemy_which_can_be_shot_by_trap(const struct
                         {
                             if (line_of_sight_2d(&traptng->mappos, &thing->mappos))
                             {
-                                // This function should return max value when the distance is minimal, so:
-                                return LONG_MAX - distance;
+                                if (creature_affected_by_spell(thing, SplK_Invisibility))
+                                {
+                                    struct TrapStats* trapstat = &game.conf.trap_stats[traptng->model];
+                                    if (trapstat->can_detect_invisible != 0)
+                                    {
+                                        // This function should return max value when the distance is minimal.
+                                        return LONG_MAX - distance;
+                                    }
+                                } else {
```

Comment:
no need for a structure with an else. You can return -1 when the creature is invisible and the trap cannot see invisible, and the the other return if not.

Reply:
I'll push something. Easier than explaining. Revert it if you do not like it.

Reply:
> > I'll push something. Easier than explaining. Revert it if you do not like it.
> 
> Looks better yes. Should I do the same on thing_trapcs.c then?

sure